src/platform/packages/shared/kbn-object-utils/src/flatten_object.test.ts

```diff
@@ -29,4 +29,25 @@ describe('flattenObject', () => {
       beta: 3,
     });
   });
+
+  it('should flat nested object properties as last ones (more nested ovverride already flattened)', () => {
+    const flattened = flattenObject({
+      alpha: {
```

Comment:
```suggestion
      'alpha.beta' : 99,
      alpha: {
```

Maybe to show that key ordering really does not matter?